src/app/dataset-view/dataset.component.ts

```diff
@@ -115,21 +109,12 @@ export class DatasetComponent implements OnInit, AfterContentInit, OnDestroy {
         ];
 
     /* eslint-disable  @typescript-eslint/no-explicit-any */
-    public tableData: {
-        isTableHeader: boolean;
-        displayedColumns?: any[];
-        tableSource: any;
-        latestMetadataBlock?: MetadataBlockFragment;
-        isResultQuantity: boolean;
-        isClickableRow: boolean;
-        pageInfo: PageInfoInterface;
-        totalCount: number;
-    };
-    public searchData:
-        | SearchOverviewDatasetsInterface[]
-        | SearchHistoryInterface[] = [];
-
-    public datasetMetadata: {};
+    public tableData: DatasetViewContentInterface;
+    public datasetHistory: MetadataBlockExtended[];
```

Comment:
If `tableData` alread stores very similar fields, why do we need another group similar references?
They seem to be initialized together, i.ex.:

```
        this.datasetOverviewService.onDatasetDataChanges.subscribe((history: SearchHistoryInterface[]) => {
            this.tableData.datasetDataSource = history;
            this.searchDataDatasets = history;
        });
```